paywall/src/components/lock/NoKeyLock.js

```diff
@@ -5,64 +5,74 @@ import UnlockPropTypes from '../../propTypes'
 import { LockWrapper, LockHeader, LockBody, LockFooter } from './LockStyles'
 import BalanceProvider from '../helpers/BalanceProvider'
 import Duration from '../helpers/Duration'
+import { UNLIMITED_KEYS_COUNT } from '../../constants'
 
 export const NoKeyLock = ({
+  account,
   lock,
   disabled,
   purchaseKey,
   lockKey,
-  soldOut,
-  tooExpensive,
-}) => (
-  <Wrapper
-    lock={lock}
-    disabled={disabled}
-    onClick={() => {
-      !disabled && purchaseKey(lockKey)
-    }}
-  >
-    <LockHeader>{lock.name}</LockHeader>
-    <BalanceProvider
-      amount={lock.keyPrice}
-      render={(ethPrice, fiatPrice) => (
-        <div>
-          <Body disabled={disabled}>
-            <EthPrice>{ethPrice} Eth</EthPrice>
-            <div>
-              <FiatPrice>${fiatPrice}</FiatPrice>
-              <Separator> | </Separator>
-              <ExpirationDuration>
-                <Duration seconds={lock.expirationDuration} round />
-              </ExpirationDuration>
-            </div>
-            <Footer>
-              {soldOut
-                ? 'Sold Out'
-                : tooExpensive
-                ? 'Insufficient funds'
-                : 'Purchase'}
-            </Footer>
-          </Body>
-        </div>
-      )}
-    />
-  </Wrapper>
-)
+}) => {
+  const soldOut =
+    lock.outstandingKeys >= lock.maxNumberOfKeys &&
+    lock.maxNumberOfKeys !== UNLIMITED_KEYS_COUNT
+
+  // When the lock is not disabled for other reasons (pending key on
+  // other lock...), we need to ensure that the lock is disabled
+  // when the lock is sold out or too expensive for the current account
+  const tooExpensive =
+    account && parseFloat(account.balance) <= parseFloat(lock.keyPrice)
+
+  return (
+    <Wrapper
+      lock={lock}
+      disabled={disabled || tooExpensive || soldOut}
+      onClick={() => {
+        !disabled && purchaseKey(lockKey)
+      }}
+    >
+      <LockHeader>{lock.name}</LockHeader>
+      <BalanceProvider
+        amount={lock.keyPrice}
+        render={(ethPrice, fiatPrice) => (
+          <div>
+            <Body disabled={disabled}>
+              <EthPrice>{ethPrice} Eth</EthPrice>
+              <div>
+                <FiatPrice>${fiatPrice}</FiatPrice>
+                <Separator> | </Separator>
+                <ExpirationDuration>
+                  <Duration seconds={lock.expirationDuration} round />
+                </ExpirationDuration>
+              </div>
+              <Footer>
+                {soldOut
+                  ? 'Sold Out'
```

Comment:
might be an opportunity to pull this out to the top of the component

Reply:
not sure I understand. This was _actually_ pulled from the top component in here... because here is the only place where it makes sense to show?

Reply:
again, I was unclear, I mean:

```js
let footerMessage
if (soldOut) {
  footerMessage = 'Sold Out'
} else if (tooExpensive) {
  footerMessage = 'Insufficient funds'
} else {
  footerMessage = 'Purchase'
}
```

Reply:
and then use `footerMessage` in the component stuff